Catch error and return null when findById/findOne doesn't find an entity

angfal opened the pull request:
### Description

When calling `findOne` or `findById` on an Model which doesn't have an entity, the default behaviour on loopback is to return `null`. When calling these methods on a Model that uses a remote, it does not return `null` but rejects the Promise. To adjust these to the official loopback models the callback is wrapped in this case to return null on `model_not_found` error.

**The description and basic code was taken from #91. Thank you, @joetjengerdes, to raise the problem**

#### Related issues

<!--
Please use the following link syntaxes:

- connect to #49 (to reference issues in the current repository)
- connect to strongloop/loopback#49 (to reference issues in another repository)
-->

- connect to #91 

### Checklist

<!--
- Please mark your choice with an "x" (i.e. [x], see
https://github.com/blog/1375-task-lists-in-gfm-issues-pulls-comments)
- PR's without test coverage will be closed.
-->

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

slnode commented:
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

angfal commented:
@bajtos I fixed #91 and added the tests. The pull requests have formatting problems, but I can't resolve it as no access to details. Could you please help with it?

bajtos commented:
@slnode ok to test

angfal commented:
@bajtos Thank you for fixing

bajtos commented:
Our CLAHUB check has been fixed and all is green, let's get this landed.

bajtos commented:
Landed, thank you @angfal for the contribution!
